web/assets/js/actions.ts

```diff
@@ -1,214 +1,391 @@
+    }
+
+    /**
+     * Fetches the Actions instance. If no instance is
+     * initialized, it initializes it
+     */
+    public static async getInstance(): Promise<Actions> {
+        if (!Actions.instance) {
+            Actions.instance = new Actions()
+
+            await Actions.instance.initialize()
+        }
+
+        return Actions.instance
+    }
+
+    /**
+     * Prepares the Actions instance
+     * @private
+     */
+    private async initialize() {
+        // Wallet initialization //
+
+        // Try to load the mnemonic from local storage
+        let mnemonic: string | null = localStorage.getItem(defaultMnemonicKey)
+        if (!mnemonic || mnemonic === "") {
+            // Generate a fresh mnemonic
+            mnemonic = generateMnemonic()
+
+            // Save the mnemonic to local storage
+            saveToLocalStorage(defaultMnemonicKey, mnemonic)
+        }
+
+        // Initialize the wallet using the saved mnemonic
+        this.wallet = await GnoWallet.fromMnemonic(mnemonic);
+
+        // Initialize the provider
+        this.provider = new GnoWSProvider(wsURL)
+
+        // Connect the wallet to the provider
+        this.wallet.connect(this.provider)
+
+        // Faucet token initialization //
+        let faucetToken: string | null = localStorage.getItem(defaultFaucetTokenKey)
+        if (faucetToken && faucetToken !== "") {
+            // Faucet token initialized
+            this.faucetToken = faucetToken
+        }
+    }
+
+    /**
+     * Saves the faucet token to local storage
+     * @param token the faucet token
+     */
+    public setFaucetToken(token: string) {
+        this.faucetToken = token
+
+        localStorage.setItem(defaultFaucetTokenKey, token)
+    }
+
+    /**
+     * Fetches the saved faucet token, if any
+     */
+    public getFaucetToken(): string | null {
+        return this.faucetToken
+    }
+
+    /****************
+     * GAME ENGINE
+     ****************/
+
+    /**
+     * Joins the waiting lobby for the game
+     * @param time
+     */
+    public async joinLobby(time: GameTime): Promise<GameSettings> {
+        // Join the waiting lobby
+        const joinResponse: BroadcastTxCommitResult = await this.wallet?.callMethod(
+            chessRealm,
+            "JoinLobby", // TODO change when API is ready
+            [
+                time.time.toString(),
+                time.increment.toString()
+            ],
+            TransactionEndpoint.BROADCAST_TX_COMMIT
+        ) as BroadcastTxCommitResult
+
+        // Parse the response from the node
+        const joinDataRaw: string | null = joinResponse.deliver_tx.ResponseBase.Data
+        if (!joinDataRaw) {
+            throw new Error("invalid join lobby response")
+        }
+
+        // Magically parse the response
+        const joinData: GamePromise = JSON.parse(joinDataRaw)
+
+        // Wait to be matched with an opponent
+        return await this.waitForGame(joinData.id)
```

Comment:
Does it return the gameID?

Reply:
It will yes, I've put down GameSettings for now (which is not accurate), as the API for JoinLobby is still not finished